src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSigner.cs

```diff
@@ -258,14 +258,31 @@ internal void CheckCertificateValue()
                     X509Chain chain = new X509Chain();
                     chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
                     chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
+                    chain.ChainPolicy.VerificationTime = Certificate!.NotBefore;
 
                     if (!chain.Build(Certificate!))
                     {
                         foreach (X509ChainStatus status in chain.ChainStatus)
                         {
                             if (status.Status == X509ChainStatusFlags.PartialChain)
                             {
+#if NET6_0_OR_GREATER
```

Comment:
Hmm... this is problematic.  System.Security.Cryptography.Pkcs is distributed via NuGet, and we don't currently have a net6.0 lib (we let it use lower netcoreappsomething).

Maybe just change the if to be the assert condition, namely, `if (chain.ChainElements.Count == 0)`, then the comment that this happens in Android when (etc)